js/shared.js

```diff
@@ -2181,3 +2246,19 @@ export const getFirebaseUI = async () => {
         }
     }
 }
+
+/**
+ * Set a short timeout to disallow multiple repetitive actions (button clicks).
+ * @param {Function} func - Function to execute.
+ * @param {number} delay - Delay in milliseconds.
+ * @returns 
+ */
+export const debounceClick = (func, delay) => {
```

Comment:
Based on my understanding, this function delays function runs but doesn't block them. So, for the use case, subsequent clicks (after first click) may not be blocked.

Reply:
I updated to toggling btn.disabled true/false instead. Plus that method doesn't introduce any unnecessary delays.